src/Storages/MergeTree/MergeFromLogEntryTask.cpp

```diff
@@ -94,7 +94,11 @@ ReplicatedMergeMutateTaskBase::PrepareResult MergeFromLogEntryTask::prepare()
         if (!source_part_or_covering)
         {
             /// We do not have one of source parts locally, try to take some already merged part from someone.
-            LOG_DEBUG(log, "Don't have all parts (at least part {} is missing) for merge {}; will try to fetch it instead", source_part_name, entry.new_part_name);
+            LOG_DEBUG(log, "Don't have all parts "
+                "(at least part {} is missing, "
+                "likely it has not been fetched yet, "
+                "maybe you pool for fetches is straving, see background_fetches_pool_size) "
```

Comment:
Or maybe some part is missing, and no active replica has it (but some inactive one has)

Reply:
Yeah. The idea was to **include only the most likely case**, since there **can be lots of such messages in the logs**, and if it will be as verbose as `Data after merge is not byte-identical to data on another replicas` then it **may be more harm then benefit**. What do you think?

So I'm suggesting one of the following:

```c
/// We do not have one of source parts locally, try to take some already merged part from someone.
LOG_DEBUG(log, "Don't have all parts "
    "(at least part {} is missing, "
    "likely it has not been fetched yet, "
    "maybe your pool for fetches is starving, see background_fetches_pool_size) "
    "for merge {}; will try to fetch it instead", source_part_name, entry.new_part_name);
```

Or

```
/// We do not have one of source parts locally, try to take some already merged part from someone.
LOG_DEBUG(log, "Don't have all parts for merge {}; will try to fetch it instead. "
    "At least part {} is missing, possible reasons are: "
    "- Likely it has not been fetched yet (maybe your pool for fetches is starving, see background_fetches_pool_size). "
    "- Only inactive replica has it. "
    "- Part is missing (maybe this replica was offline for too long and the part had been removed already from all replicas, or it lost forever). "
    entry.new_part_name,
    source_part_name);
```

Thoughts?

Reply:
Maybe something like this:
```
LOG_DEBUG(log, "Don't have all parts "
    "(at least part {} is missing, "
    "likely it has not been fetched yet)"
    "for merge {}; will try to fetch it instead. "
    "Maybe the part is missing or the pool for fetches is starving, "
    "see background_fetches_pool_size",
    source_part_name, entry.new_part_name);
```

Reply:
Went with 

```c
LOG_DEBUG(log, "Don't have all parts (at least {} is missing) for merge {}; will try to fetch it instead. "
    "Either pool for fetches is starving, see background_fetches_pool_size, or none of active replicas has it",
   source_part_name, entry.new_part_name);
```

How that sounds to you?